clang/lib/CodeGen/CodeGenPGO.cpp

```diff
@@ -292,7 +295,7 @@ struct MapRegionCounters : public RecursiveASTVisitor<MapRegionCounters> {
                 "contains an operation with a nested boolean expression. "
                 "Expression will not be covered");
             Diag.Report(S->getBeginLoc(), DiagID);
-            return false;
+            return true;
```

Comment:
Actually my point is: returning `false` would exit early from the AST traversal, and I suppose both places *have to* return `true`. Otherwise we'd observe things in my original post.

Reply:
Yes, that's what I mean.  There was no benefit in returning false to begin with; you have to return true.  The checking that the pass does to catch the case, issue the diagnostic, and avoid processing of the data for the expression is sufficient to handle the case.